web/client/epics/maps.js

```diff
@@ -350,12 +356,82 @@ const createMapResource = (resource) => Persistence.createResource(resource)
     ))
     .startWith(savingMap(resource.metadata));
 /**
- * Create or update map reosurce with persistence api
+ * Create or update map resource with persistence api
  */
 const mapSaveMapResourceEpic = (action$) =>
     action$.ofType(SAVE_MAP_RESOURCE)
-        .exhaustMap(({resource}) => (!resource.id ? createMapResource(resource) : updateMapResource(resource))
-        );
+        .exhaustMap(({resource}) => (!resource.id ? createMapResource(resource) : updateMapResource(resource)));
+
+const updateMap = (response, backgroundID) => {
+    const thumbnailUrl = ConfigUtils.getDefaults().geoStoreUrl + "data/" + response + "/raw?decode=datauri";
+    // add encodedThumbnailUrl and id to the background thumbnail source
+    return Rx.Observable.of(
+        updateNode(backgroundID, "layers", { thumbURL: thumbnailUrl, thumbId: response }),
+        updateThumbnail(null, null, backgroundID));
+};
+
+const createBackgroundThumbnail = (thumbnail, categoryThumbnail, backgroundID) => {
+    if (thumbnail && thumbnail.url && thumbnail.data) {
+        const metadata = {
+            name: thumbnail.url
+        };
+
+        return Persistence.createResource({metadata: metadata, data: thumbnail.data, category: categoryThumbnail, backgroundID})
+            .map((response) => ({response, backgroundID}));
+    }
+
+    return Rx.Observable.of(null);
+};
+
+const mapSaveBackgroundThumbnails = (action$) =>
+    action$.ofType(TRIGGER_SAVE_MAP)
+        .switchMap((action) => {
+            // list of source id's of backgrounds with removed thumbnails
+            const resourcesToRemove = action.data.backgrounds
+                .filter(background => background.thumbnail && background.thumbnail.data && background.thumbId)
+                .map(background => background.thumbId);
+            return Rx.Observable.forkJoin(Rx.Observable.of(null),
+                ...action.data.backgrounds.map(layer =>
+                    createBackgroundThumbnail(layer.thumbnail, "BACKGROUND_THUMBNAIL", layer.id))
```

Comment:
Hi, this epic needs some refactor or be removed. The new requirements were to save and compress background thumbnails in the map configuration, so **all the parts** where is being saved a thumbnail in geostore is not needed anymore.  please check and rewrite al the involved parts